include/dali/core/geom/mat.h

```diff
@@ -481,6 +481,34 @@ mat<r1+r2, cols, T> cat_rows(const mat<r1, cols, T> &a, const mat<r2, cols, T> &
   return ret;
 }
 
+
+/**
+ * Returns a dali::mat with ones on the diagonal and zeros elsewhere.
+ */
+template <int rows, int cols, typename T = float>
+DALI_HOST_DEV constexpr dali::mat<rows, cols, T> eye() {
+  mat<rows, cols, T> ret(0);
+  for (int i = 0; i < rows && i < cols; i++) {
+    ret(i, i) = 1;
+  }
+  return ret;
+}
+
+
+template <int N, typename T = float>
+DALI_HOST_DEV constexpr dali::mat<N, N, T> eye() {
+  return eye<N, N, T>();
+}
+
```

Comment:
These functions are redundant. Construction from scalar puts that scalar on the diagonal.

Reply:
This is in my opinion baaaad. Is there any FW or library, that does the diagonal-initialization by the constructor? I'd rather expect the construtor to initialize all the matrix elements with given value (cf. [OpenCV](https://docs.opencv.org/trunk/d3/d63/classcv_1_1Mat.html#a3620c370690b5ca4d40c767be6fb4ceb)). I'd vote for chaning the behavoiur of the constructor, but not in this PR, since this change can create problems...

Reply:
That's what you have in OpenGL / shader programming.
Since matrices are usually used for multiplaction and vectors for addition, this gives consistent results:
```
mat3 A;
A * 5 == A * mat3(5);
vec4 v;
v + 5 = v + vec4(5);
```